File: 04 자바기초/source/day01/src/exam03/Ex06.java
package exam03;

public class Ex06 {
    public static void main(String[] args) {
        int num1 = 1000000000;// 정수면 무조건 int부터 자료형을 결정 -> int가 아니면 long으로 변환

        long num2 = 10000000000L; // 기본적으로 처음에는 long 선언 후에도 int로 인식하고 있어서 오류 처리가 적용됨 -> L을 뒤에 붙여 처음부터 long으로 선언하면 간단 처리~!
        long num3 = 10_000_000_000L; // 가독성을 높히기 위한 선언 방법
    }
}

File: js_class/js/ex06.js
// 가변 길이 인수 목록

function sum(){
    let total = 0;
    for(const num of arguments){
        total += num;
    }
    console.log(total);
    return total;
}

sum(1, 2, 3);

function sum1(...args){
    let total = 0;
    for(const num of args){
        total += num;
    }
    console.log(total);
    return total;
}

sum1(10, 20, 30);

function sum2(num1, num2, ...params){
    console.log("num1", num1, "num2", num2);
    console.log("params", params);
}

sum2(10, 20, 30, 40, 50);